pallets/settlement/src/benchmarking.rs

```diff
@@ -473,94 +786,139 @@ benchmarks! {
+        }
+    }
+
+
+    claim_receipt {
+        // There is no catalyst in this dispatchable, It will always be time constant.
+        // Emulate the add instruction and get all the necessary arguments.
+        let (legs, venue_id, origin, did , s_portfolios, r_portfolios, account_id) = emulate_add_instruction::<T>(1, true)?;
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id = 1;
+        let ticker = Ticker::try_from(vec![b'A'; 1 as usize].as_slice()).unwrap();
+        let receipt = create_receipt_details::<T>(0, legs.first().unwrap().clone());
+        let leg_id = 0;
+        let amount = 100;
+        // Some manual setup to support the extrinsic.
+        set_instruction_leg_status_to_pending::<T>(instruction_id, leg_id);
+        T::Portfolio::lock_tokens(s_portfolios.first().unwrap(), &ticker, &amount.into())?;
+    }: _(origin, instruction_id, receipt.clone())
+    verify {
+        ensure!(Module::<T>::instruction_leg_status(instruction_id, leg_id) ==  LegStatus::ExecutionToBeSkipped(
+            receipt.signer,
+            receipt.receipt_uid,
+        ), "Settlement: Fail to unclaim the receipt");
+    }
+
+
+    affirm_with_receipts {
+        // Catalyst here is the length of receipts vector.
+        let r in 1 .. T::MaxLegsInInstruction::get() as u32;
+        // Emulate the add instruction and get all the necessary arguments.
+        let (legs, venue_id, origin, did , s_portfolios, r_portfolios, account_id) = emulate_add_instruction::<T>(r, true)?;
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id = 1;
+        let mut receipt_details = Vec::with_capacity(r as usize);
+        legs.clone().into_iter().enumerate().for_each(|(idx, l)| {
+            receipt_details.push(create_receipt_details::<T>(idx as u32, l));
+        });
+    }: _(origin, instruction_id, receipt_details.clone(), s_portfolios)
+    verify {
+        for (i, receipt) in receipt_details.iter().enumerate() {
+            ensure!(Module::<T>::instruction_leg_status(instruction_id, i as u64) ==  LegStatus::ExecutionToBeSkipped(
+                receipt.signer.clone(),
+                receipt.receipt_uid,
+            ), "Settlement: Fail to affirm with receipts");
+        }
+    }
+
+    execute_scheduled_instruction {
+        // This dispatch execute an instruction.
+        //
+        // Worst case scenarios.
+        // 1. Create maximum legs and both traded assets are different assets/securities.
+        // 2. Assets should have worst compliance restrictions ?
+        // 3. Assets have maximum no. of TMs.
+
+        let l in 2 .. T::MaxLegsInInstruction::get() as u32; // At least 2 legs needed to achieve worst case.
```

Comment:
Why do we want at least 2 legs? IMO having even 0 legs is fine, it scales properly w.r.t number of legs.

If we benchmark with min 2, the cost of 0 legs might come out to be 0 since the weight function will think 0/1 legs are impossible.

Reply:
I don't think it will affect in any way. 0 legs mean no instruction at all so it automatically turns out to be 0 weight for that. And if we talk about the 1 legs then it will be direct multiplication with the no. of legs so I don't think weight function doesn't work properly if we pass 1 leg.

Reply:
>  0 legs mean no instruction at all so it automatically turns out to be 0 weight for that.

No. Even if there are 0 legs, there is work to be done.

> And if we talk about the 1 legs then it will be direct multiplication with the no. of legs so I don't think weight function doesn't work properly if we pass 1 leg.

Except that the benchmarking CLI won't have proper scaling numbers. Please just set it to 0 and let the CLI do its magic. No need to over engineer.

Reply:
`No. Even if there are 0 legs, there is work to be done.`

then I think we have to restrict the 0 leg instruction, There is no sense in having instruction with 0 legs.

Reply:
Even if we restrict it, that doesn't mean there's no work to be done....

You need some weight to fail the tx.

Reply:
Ture we have that weight if we read anything but related to legs it will be 0.